### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/fwupd/commit/c3a64ed78e03a0ab2ba230bdf83d8cc48776519f#commitcomment-94360247

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/fwupd/commit/8d80af781a84a8fc965b5f07229fafd7d29cb463

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: contrib/qubes/doc/heads_update.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ In the main menu, choose `options` and then go to `Flash/Update the BIOS`
 
 Decide to retain or erase the settings.
 
-![img](img/heads_firmware_managment_menu.jpg)
+![img](img/heads_firmware_management_menu.jpg)
```
Comment (jsoref) [inline contrib/qubes/doc/heads_update.md:31]: Oops, missed a file rename.

File: data/org.freedesktop.fwupd.metainfo.xml [hunk 2]
```diff
@@ -557,7 +557,7 @@
         <p>This release fixes the following bugs:</p>
         <ul>
           <li>Be case insensitive when fixing the device model</li>
-          <li>Fix a critial warning in ccgx found by the fuzzer</li>
+          <li>Fix a critical warning in ccgx found by the fuzzer</li>
```
Comment (jsoref) [inline data/org.freedesktop.fwupd.metainfo.xml:560]: Some projects don't like changing changelogs. Happy to drop this or anything else.

File: docs/hsi-tests.d/org.fwupd.hsi.PlatformDebugLocked.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "If using a debug unlocked part, the platform's overall security will be decreased as an attacker may have elevated access to registers and memory within the system and can potentially enable persistent backdoors."
   ],
   "failure-results" : {
-    "not-locked" : "device is not not locked"
+    "not-locked" : "device is not locked"
```
Comment (jsoref) [inline docs/hsi-tests.d/org.fwupd.hsi.PlatformDebugLocked.json:15]: double negatives can be very confusing

File: libfwupdplugin/fu-fit-firmware.c [hunk 4]
```diff
@@ -244,7 +244,7 @@ fu_fit_firmware_verify_image(FuFirmware *firmware,
 	if (g_getenv("FU_FDT_FIRMWARE_VERBOSE") != NULL)
 		fu_dump_bytes(G_LOG_DOMAIN, "data", blob);
 
-	/* verify any hashes we recoginise */
+	/* verify any hashes we recognize */
```
Comment (jsoref) [inline libfwupdplugin/fu-fit-firmware.c:247]: Google picked the `-ize` ending (and I consciously chose not to intervene). If this project has a strong `-ise` preference, I can certainly fix that.

File: libfwupdplugin/fu-usb-device-ms-ds20.c [hunk 5]
```diff
@@ -18,7 +18,7 @@ G_DEFINE_TYPE(FuUsbDeviceMsDs20, fu_usb_device_ms_ds20, FU_TYPE_USB_DEVICE_DS20)
 #define USB_OS_20_SET_HEADER_DESCRIPTOR	      0x00
 #define USB_OS_20_SUBSET_HEADER_CONFIGURATION 0x01
 #define USB_OS_20_SUBSET_HEADER_FUNCTION      0x02
-#define USB_OS_20_FEATURE_COMPATBLE_ID	      0x03
+#define USB_OS_20_FEATURE_COMPATIBLE_ID	      0x03
```
Comment (jsoref) [inline libfwupdplugin/fu-usb-device-ms-ds20.c:21]: I've generally tried to retain whitespace alignment.

Some projects don't like changing defines, but this is a `.c` so it should be a local and harmless to rename.

File: libfwupdplugin/fu-usb-device-ms-ds20.c [hunk 6]
```diff
@@ -34,8 +34,8 @@ fu_usb_device_os20_type_to_string(guint16 type)
 		return "subset-header-configuration";
 	if (type == USB_OS_20_SUBSET_HEADER_FUNCTION)
 		return "subset-header-function";
-	if (type == USB_OS_20_FEATURE_COMPATBLE_ID)
-		return "feature-compatble-id";
+	if (type == USB_OS_20_FEATURE_COMPATIBLE_ID)
+		return "feature-compatible-id";
```
Comment (jsoref) [inline libfwupdplugin/fu-usb-device-ms-ds20.c:38]: This change is more interesting. As something might care about the returned string.

File: libfwupdplugin/tests/bios-attrs/dell-xps13-9310/dell-wmi-sysman/attributes/EmbSataRaid/display_name [hunk 7]
```diff
@@ -1 +1 @@
-SATA/NVMe Opeartion
+SATA/NVMe Operation
```
Comment (jsoref) [inline libfwupdplugin/tests/bios-attrs/dell-xps13-9310/dell-wmi-sysman/attributes/EmbSataRaid/display_name:1]: I believe this is human readable and not tied to the underlying value, but I could be very wrong.

File: plugins/fresco-pd/lsusb.txt [hunk 8]
```diff
@@ -55,7 +55,7 @@ Binary Object Store Descriptor:
     bLength                    48
     bDescriptorType            16
     bDevCapabilityType         13
-    iAddtionalInfoURL           4 www.frescologic.com
+    iAdditionalInfoURL          4 www.frescologic.com
```
Comment (jsoref) [inline plugins/fresco-pd/lsusb.txt:58]: ?

File: plugins/nitrokey/lsusb.txt [hunk 9]
```diff
@@ -72,7 +72,7 @@ Device Descriptor:
         bVoltageSupport         2  3.0V 
         dwProtocols             2  T=1
         dwDefaultClock       3600
-        dwMaxiumumClock      3600
+        dwMaximumClock      3600
```
Comment (jsoref) [inline plugins/nitrokey/lsusb.txt:75]: Oops. Missed.

File: plugins/synaptics-rmi/fu-synaptics-rmi-hid-device.c [hunk 10]
```diff
@@ -41,7 +41,7 @@ G_DEFINE_TYPE(FuSynapticsRmiHidDevice, fu_synaptics_rmi_hid_device, FU_TYPE_SYNA
 #define HID_RMI4_WRITE_OUTPUT_ADDR     2
 #define HID_RMI4_WRITE_OUTPUT_DATA     4
 #define HID_RMI4_FEATURE_MODE	       1
-#define HID_RMI4_ATTN_INTERUPT_SOURCES 1
+#define HID_RMI4_ATTN_INTERRUPT_SOURCES 1
```
Comment (jsoref) [inline plugins/synaptics-rmi/fu-synaptics-rmi-hid-device.c:44]: I could fix all the adjacent items...

File: po/ca.po [hunk 11]
```diff
@@ -2790,7 +2790,7 @@ msgstr "Fallada transitòria"
 msgid "True"
 msgstr "Cert"
 
-#. TRANSLATORS: We verified the meatdata against the server
+#. TRANSLATORS: We verified the metadata against the server
```
Comment (jsoref) [inline po/ca.po:2793]: Some projects don't like changes to localization files... But it's about 50/50. (This was unintentional, I had excluded this file, so it shouldn't have been changed...)

File: src/fu-self-test.c [hunk 12]
```diff
@@ -4930,7 +4930,7 @@ main(int argc, char **argv)
 	g_test_add_data_func("/fwupd/plugin-list", self, fu_plugin_list_func);
 	g_test_add_data_func("/fwupd/plugin-list{depsolve}", self, fu_plugin_list_depsolve_func);
 	g_test_add_func("/fwupd/spawn", fu_spawn_func);
-	g_test_add_func("/fwupd/spawn-timeou)", fu_spawn_timeout_func);
+	g_test_add_func("/fwupd/spawn-timeout", fu_spawn_timeout_func);
```
Comment (jsoref) [inline src/fu-self-test.c:4933]: notable

**Review by jsoref — COMMENTED** [review]
Most changes suggested by Google Sheets. All fault mine.

File: data/org.freedesktop.fwupd.metainfo.xml (hunk 2 above)
Comment (hughsie) [reply data/org.freedesktop.fwupd.metainfo.xml:560]: It's fine for me!

File: libfwupdplugin/fu-fit-firmware.c (hunk 4 above)
Comment (hughsie) [reply libfwupdplugin/fu-fit-firmware.c:247]: Although I'm en_GB, I think we use en_US everywhere

File: libfwupdplugin/fu-udev-device.c [hunk 13]
```diff
@@ -1000,7 +1000,7 @@ fu_udev_device_get_vendor(FuUdevDevice *self)
  * fu_udev_device_get_model:
  * @self: a #FuUdevDevice
  *
- * Gets the device device code.
+ * Gets the device code.
```
Comment (hughsie) [inline libfwupdplugin/fu-udev-device.c:1003]: Should be 'device model code'

File: libfwupdplugin/fu-usb-device-ms-ds20.c (hunk 5 above)
Comment (hughsie) [reply libfwupdplugin/fu-usb-device-ms-ds20.c:21]: Just do what clang-format wants please

File: libfwupdplugin/fu-usb-device-ms-ds20.c (hunk 6 above)
Comment (hughsie) [reply libfwupdplugin/fu-usb-device-ms-ds20.c:38]: I think this instance it's fine to change

File: libfwupdplugin/tests/bios-attrs/dell-xps13-9310/dell-wmi-sysman/attributes/EmbSataRaid/display_name (hunk 7 above)
Comment (hughsie) [reply libfwupdplugin/tests/bios-attrs/dell-xps13-9310/dell-wmi-sysman/attributes/EmbSataRaid/display_name:1]: @superm1 ?

File: plugins/dfu/fu-dfu-target-stm.c [hunk 14]
```diff
@@ -298,7 +298,7 @@ fu_dfu_target_stm_download_element1(FuDfuTarget *target,
 			}
 
 			/* if it's erasable and not yet blanked */
-			if (fu_dfu_sector_has_cap(sector, DFU_SECTOR_CAP_ERASEABLE) &&
+			if (fu_dfu_sector_has_cap(sector, DFU_SECTOR_CAP_ERASABLE) &&
```
Comment (hughsie) [inline plugins/dfu/fu-dfu-target-stm.c:301]: Is that correct?

File: plugins/fresco-pd/lsusb.txt (hunk 8 above)
Comment (hughsie) [reply plugins/fresco-pd/lsusb.txt:58]: Looks like libusb needs the same checks :) this is fine for me

File: plugins/synaptics-rmi/fu-synaptics-rmi-hid-device.c (hunk 10 above)
Comment (hughsie) [reply plugins/synaptics-rmi/fu-synaptics-rmi-hid-device.c:44]: Clang format please

File: po/ca.po (hunk 11 above)
Comment (hughsie) [reply po/ca.po:2793]: Maybe don't include .po please

File: libfwupdplugin/tests/bios-attrs/dell-xps13-9310/dell-wmi-sysman/attributes/EmbSataRaid/display_name (hunk 7 above)
Comment (superm1) [reply libfwupdplugin/tests/bios-attrs/dell-xps13-9310/dell-wmi-sysman/attributes/EmbSataRaid/display_name:1]: This was captured from a real machine; so it's a BIOS bug.  We shouldn't change it.

File: plugins/dfu/fu-dfu-target-stm.c (hunk 14 above)
Comment (jsoref) [reply plugins/dfu/fu-dfu-target-stm.c:301]: Well, it matches the comment on line 300...

File: plugins/fresco-pd/lsusb.txt (hunk 8 above)
Comment (jsoref) [reply plugins/fresco-pd/lsusb.txt:58]: Added to https://github.com/stars/jsoref/lists/to-check-spelling, but the list doesn't do a great job of shrinking...

File: contrib/qubes/src/qubes_fwupdmgr.py [hunk 15]
```diff
@@ -128,7 +128,7 @@ def refresh_metadata(self, whonix=False, metadata_url=None):
             raise Exception("Manual metadata refresh failed!!!")
 
     def _get_dom0_updates(self):
-        """Gathers infromations about available updates."""
+        """Gathers information about available updates."""
```
Comment (jsoref) [inline contrib/qubes/src/qubes_fwupdmgr.py:131]: codespell made me do it.

(I personally prefer this spelling, but Google suggested the simpler fix and I didn't feel like being particularly opinionated.)

Comment (hughsie) [reply contrib/qubes/src/qubes_fwupdmgr.py:131]: I think information is better here.

File: plugins/dfu/fu-dfu-target-stm.c (hunk 14 above)
Comment (hughsie) [reply plugins/dfu/fu-dfu-target-stm.c:301]: You're correct, erasable is correct English.

File: src/fu-self-test.c (hunk 12 above)
Comment (hughsie) [reply src/fu-self-test.c:4933]: but good

**Comment by hughsie** [comment]
@superm1 all these look good, but do we want this as one commit or multiple? I'm erring on lots of little commits in case we need to revert one in the future.

**Comment by superm1** [comment]
> @superm1 all these look good, but do we want this as one commit or multiple? I'm erring on lots of little commits in case we need to revert one in the future.

I don't feel strongly.

**Comment by hughsie** [comment]
@jsoref many thanks for all your work!

**Pull request merged**